### ISSUE TITLE
training issue

### ISSUE THREAD

**Issue by mdqyy116** [qiexing/caffe-regression#1]
Hi guys,

Great work. I use your network trained on my data, but the loss can't reduce. Can you share your training log file and .h5 data to me. I want to do some compare.

Thanks,
Richard

**Comment by qiexing** [comment]
Hi, Richard .
you can download data from https://www.kaggle.com/c/facial-keypoints-detection/data
and use the kfkd.py under the kaggle_fkp_data folder to generate  .h5 data.

**Comment by mdqyy116** [comment]
Yes, I download the data from the kaggle. Then generate the .h5 data using your python script, but the result was not correct, I found the loss not decrease after about 5000 iterations. So I want to have a look your log file and use the data you generated. I shared my training data, log file and model file to the following link, please help me check what's the problem?
https://www.dropbox.com/sh/ajh6zs20qncrexn/AADjH2BW1dUh7NpTzNaYs9cMa?dl=0

Thanks,
Richard

**Comment by qiexing** [comment]
The result is not correct you mean the loss is not decrease to 0 ?To verify, you can use the test.h5 to test your facial point model. In this task ,the data only 1000+ and the model is simple, is an simple regression examples. To you own problem, you say in your data, but the loss can't reduce. I think you need do data augmentation.I have same data and model as you.The log file had been emptied.I had to train the model again.I will upload soon.

**Comment by mdqyy116** [comment]
Your results are quite impressive for me. I think your loss should be about 0.002, but I just can achieve 0.02. Thanks for re-training the model.

**Comment by mdqyy116** [comment]
Hi guys,

Have you finished the training? BTW, have you augmented the data before training?

Thanks,
Richard

**Comment by qiexing** [comment]
I had updated the training log. Different training tools may have little different  in computing loss. Directly calculate point error may better.
In this example, just want to show how to do regression in caffe  , without data augmentation

**Comment by mdqyy116** [comment]
Thanks, your loss make sense for me.  I think the error can compute as "error = sqrt(loss)*48";
